nucypher/crypto/kits.py

```diff
@@ -76,3 +80,78 @@ def __init__(self, *args, **kwargs) -> None:
     def from_bytes(cls, some_bytes):
         capsule, sender_pubkey_sig, ciphertext = cls.split_bytes(some_bytes)
         return cls(capsule=capsule, sender_pubkey_sig=sender_pubkey_sig, ciphertext=ciphertext)
+
+
+Revocation = namedtuple('Revocation', ['prefix', 'arrangement_id', 'signature'])
+
+
+class RevocationKit:
+
+    def __init__(self, policy_treasure_map):
+        self.revocations = dict()
+        for node_id, arrangement_id in policy_treasure_map:
+            self.revocations[node_id] = Revocation(b'REVOKE-', arrangement_id,
+                                                   constants.NOT_SIGNED)
+
+    def __iter__(self):
+        return iter(self.revocations.values())
+
+    def __getitem__(self, node_id):
+        return self.revocations[node_id]
+
+    def __len__(self):
+        return len(self.revocations)
+
+    def __eq__(self, other):
+        return self.revocations == other.revocations
+
+    @staticmethod
+    def revocation_to_bytes(revocation):
+        prefix, arrangement_id, signature = revocation
+        return b''.join((prefix, arrangement_id, bytes(signature)))
+
+    @staticmethod
+    def revocation_from_bytes(revocation_as_bytes: bytes):
+        from nucypher.crypto.signing import Signature
+        revocation_splitter = BytestringSplitter((bytes, 7),
+                                                 (bytes, 32),
+                                                 Signature)
```

Comment:
This fails when revocations haven't been signed, as `Signature` is expected to be 64 bytes, and `constants.NOT_SIGNED` is just 4 bytes.

Reply:
This I know, the problem is that serializing unsigned revocations makes little to no sense. It hardly seems worth the complexity to build such a thing. You can construct it with a copy of the treasure map with no effort.

Reply:
Perhaps a conditional to handle the case @cygnusv described?  or am I misunderstanding something?

Reply:
I think you're misunderstanding it a bit. Serializing unsigned revocations is _absolutely useless_. The treasure map is used to construct the revocation requests. If the revocation is unsigned, there is no point in actually serializing it at all. It's better to serialize a treasure map, then generate the revocation.

Reply:
My point is that I would rather not even include such an interface. The proper way to do this is to use the Treasure Map to generate those revocations.

Reply:
OK - Got it.  Nonetheless - a crash is possible here if `signature` is a `_Constant`, right?

Reply:
Yes, how should it be handled? I'm presuming in a similar fashion to how we handle malformed input to other BSS calls?

Reply:
Yeah - I'd regard this as a sensitive area not - from our experience with BSS - its important to have a very informative error message here.  What about preventing malformed transmissions altogether? Do you think its bad to look before you leap here (or in other similar situations) @jMyles?

Reply:
Help me understand - is there a useful function for unsigned revocations?

Reply:
They exist purely as a in-between stage for revocations that need to be signed. Once they're signed, you could serialize them and distribute them to someone, if needed.

Other than that, they don't have a function. Considering that signed revocations can be used to cancel a policy, I think of them as a not-so-toxic-yet-toxic material that shouldn't really be held unless you intend to store them or distribute them.